internal/usecase/interactor/user.go

```diff
@@ -164,6 +164,18 @@ func (i *User) Signup(ctx context.Context, inp interfaces.SignupParam) (u *user.
 	return u, team, nil
 }
 
+func (i *User) GetUserByCredentials(ctx context.Context, inp interfaces.GetUserByCredentials) (u *user.User, err error) {
+	u, err = i.userRepo.FindByNameOrEmail(ctx, inp.Email)
+	if err != nil && !errors.Is(err, rerror.ErrNotFound) {
+		return nil, err
+	}
```

Comment:
```suggestion
	}
	if u == nil {
		return nil, interfaces.ErrInvalidUserCredentials
	}
```

It's important for security to return the same error even when user is not found.